madara/crates/client/db/src/block_db.rs

```diff
@@ -355,6 +357,68 @@ impl MadaraBackend {
         Ok(())
     }
 
+    /// Reverts the tip of the chain back to the given block.
+    ///
+    /// In addition, this removes all historical data (chain state, transactions, state diffs,
+    /// etc.) from the database. `ROW_SYNC_TIP` is set to the new tip.
+    ///
+    /// Returns a Vec of `(block_number, state_diff)` where the Vec is in reverse order (the first
+    /// element is the current tip of the chain and the last is `revert_to`).
+    pub(crate) fn block_db_revert(&self, revert_to: u64) -> Result<Vec<(u64, StateDiff)>> {
+        let mut tx = WriteBatchWithTransaction::default();
+
+        let tx_hash_to_block_n = self.db.get_column(Column::TxHashToBlockN);
+        let block_hash_to_block_n = self.db.get_column(Column::BlockHashToBlockN);
+        let block_n_to_block = self.db.get_column(Column::BlockNToBlockInfo);
+        let block_n_to_block_inner = self.db.get_column(Column::BlockNToBlockInner);
+        let block_n_to_state_diff = self.db.get_column(Column::BlockNToStateDiff);
+        let meta = self.db.get_column(Column::BlockStorageMeta);
+
+        // clear pending
+        tx.delete_cf(&meta, ROW_PENDING_INFO);
+        tx.delete_cf(&meta, ROW_PENDING_INNER);
+        tx.delete_cf(&meta, ROW_PENDING_STATE_UPDATE);
```

Comment:
we also need to clear the pending block in classdb and blockdb, but I think it's just easier to remove the clearing of pending block from here and just call the .clear_pending_block() fn in the main db revert_to() function

Reply:
Removed this (since `clear_pending_block()` is being called now) and added some clarification about it in [6b49010](https://github.com/madara-alliance/madara/pull/296/commits/6b4901044efe1d400b6a48a5bbfc2b31812479d7)